src/System.Private.CoreLib/src/System/Enum.cs

```diff
@@ -282,262 +280,530 @@ internal static ulong ToUInt64(object value)
+            }
 
-                    case ParseFailureKind.UnhandledException:
-                        return m_innerException;
+            ReadOnlySpan<char> valueSpan = value.AsSpan().TrimStart();
+            if (valueSpan.Length == 0)
+            {
+                result = default;
+                return false;
+            }
 
-                    default:
-                        Debug.Fail("Unknown EnumParseFailure: " + m_failure);
-                        return new ArgumentException(SR.Arg_EnumValueNotFound);
-                }
+            switch (Type.GetTypeCode(typeof(TEnum)))
+            {
+                case TypeCode.SByte: return TryParseSByteEnum<TEnum>((RuntimeType)typeof(TEnum), value, valueSpan, ignoreCase, throwOnFailure: false, out result);
+                case TypeCode.Byte: return TryParseByteEnum<TEnum>((RuntimeType)typeof(TEnum), value, valueSpan, ignoreCase, throwOnFailure: false, out result);
+                case TypeCode.Int16: return TryParseInt16Enum<TEnum>((RuntimeType)typeof(TEnum), value, valueSpan, ignoreCase, throwOnFailure: false, out result);
+                case TypeCode.UInt16: return TryParseUInt16Enum<TEnum>((RuntimeType)typeof(TEnum), value, valueSpan, ignoreCase, throwOnFailure: false, out result);
+                case TypeCode.Int32: return TryParseInt32Enum<TEnum>((RuntimeType)typeof(TEnum), value, valueSpan, ignoreCase, throwOnFailure: false, out result);
+                case TypeCode.UInt32: return TryParseUInt32Enum<TEnum>((RuntimeType)typeof(TEnum), value, valueSpan, ignoreCase, throwOnFailure: false, out result);
+                case TypeCode.Int64: return TryParseInt64Enum<TEnum>((RuntimeType)typeof(TEnum), value, valueSpan, ignoreCase, throwOnFailure: false, out result);
+                case TypeCode.UInt64: return TryParseUInt64Enum<TEnum>((RuntimeType)typeof(TEnum), value, valueSpan, ignoreCase, throwOnFailure: false, out result);
+                default: return TryParseRareEnum<TEnum>((RuntimeType)typeof(TEnum), value, valueSpan, ignoreCase, throwOnFailure: false, out result);
             }
         }
 
-        public static bool TryParse(Type enumType, string value, out object result)
+        public static object Parse(Type enumType, string value) =>
+            Parse(enumType, value, false);
+
+        public static object Parse(Type enumType, string value, bool ignoreCase)
         {
-            return TryParse(enumType, value, false, out result);
+            if (enumType == null)
+            {
+                throw new ArgumentNullException(nameof(enumType));
+            }
+
+            RuntimeType rtType = enumType as RuntimeType;
+            if (rtType == null)
+            {
+                throw new ArgumentException(SR.Arg_MustBeType, nameof(enumType));
+            }
+
+            if (!rtType.IsEnum)
+            {
+                throw new ArgumentException(SR.Arg_MustBeEnum, nameof(enumType));
+            }
+
+            if (value == null)
```

Comment:
Why is this check required for Parse, but not for TryParse?

Reply:
Parse throws a different exception for value == null vs value is empty, whereas TryParse just returns false for both.  And as string.AsSpan() does the null check and just returns an empty span for it, we don't need a separate value == null check for TryParse.